project-base/tests/ShopBundle/Functional/Model/Vat/VatFacadeTest.php

```diff
@@ -42,35 +47,66 @@ class VatFacadeTest extends TransactionFunctionalTestCase
      */
     private $paymentFacade;
 
-    public function testDeleteByIdAndReplace()
+    /**
+     * @var \Shopsys\FrameworkBundle\Model\Pricing\Currency\CurrencyFacade
+     * @inject
+     */
+    private $currencyFacade;
+
+    public function testDeleteByIdAndReplaceForFirstDomain()
     {
         $em = $this->getEntityManager();
 
         $vatData = new VatData();
         $vatData->name = 'name';
         $vatData->percent = '10';
-        $vatToDelete = $this->vatFacade->create($vatData);
+        $vatToDelete = $this->vatFacade->create($vatData, Domain::FIRST_DOMAIN_ID);
+
         /** @var \Shopsys\FrameworkBundle\Model\Pricing\Vat\Vat $vatToReplaceWith */
-        $vatToReplaceWith = $this->getReference(VatDataFixture::VAT_HIGH);
+        $vatToReplaceWith = $this->getReference(sprintf('%s_%s', VatDataFixture::VAT_HIGH, Domain::FIRST_DOMAIN_ID));
+
         /** @var \Shopsys\ShopBundle\Model\Transport\Transport $transport */
         $transport = $this->getReference(TransportDataFixture::TRANSPORT_PERSONAL);
         $transportData = $this->transportDataFactory->createFromTransport($transport);
+
         /** @var \Shopsys\ShopBundle\Model\Payment\Payment $payment */
         $payment = $this->getReference(PaymentDataFixture::PAYMENT_CASH);
         $paymentData = $this->paymentDataFactory->createFromPayment($payment);
 
-        $transportData->vat = $vatToDelete;
+        $transportData->vatsIndexedByDomainId[Domain::FIRST_DOMAIN_ID] = $vatToDelete;
         $this->transportFacade->edit($transport, $transportData);
 
-        $paymentData->vat = $vatToDelete;
+        $paymentData->vatsIndexedByDomainId[Domain::FIRST_DOMAIN_ID] = $vatToDelete;
         $this->paymentFacade->edit($payment, $paymentData);
 
-        $this->vatFacade->deleteById($vatToDelete, $vatToReplaceWith);
-
         $em->refresh($transport);
         $em->refresh($payment);
 
-        $this->assertEquals($vatToReplaceWith, $transport->getVat());
-        $this->assertEquals($vatToReplaceWith, $payment->getVat());
+        $this->vatFacade->deleteById($vatToDelete, Domain::FIRST_DOMAIN_ID, $vatToReplaceWith);
+
+        $defaultCurrencyForFirstDomain = $this->currencyFacade->getDomainDefaultCurrencyByDomainId(Domain::FIRST_DOMAIN_ID);
+
+        $transportVat = $em->createQueryBuilder()
```

Comment:
I would prefer to load some transport instead of this sql that will be hard to sustain

Reply:
I agree, but this test was cursed for me. We can call a bout this

Reply:
We can :)

Reply:
Fixed. I don't know what I was doing wrong.